### DISCUSSION TITLE
I have set up Memcode on my server, however I'm getting a bad request error with google

### DISCUSSION THREAD

**Discussion by samintokyo** [lakesare/memcode#155] [General]
Hi - are there any parameters for the callback url or additional steps for setting up google signup auth on a production server?

https://cards.cotoacademy.com

Have installed here but cannot find the reason why i keep getting a bad request after a stall out on google auth.

**Comment by lakesare** [comment]
Hi, Samuel! You need to create **your own gmail oauth account** to make this work.
The gmail account from our development `env.js` redirects you to `https://localhost:3000/api/auth/google/callback`, and it should instead redirect you to `https://cards.cotoacademy.com/api/auth/google/callback`.

Here is an example of how this is set up for Memcode on Google:
<img width="606" alt="image" src="https://user-images.githubusercontent.com/7578559/156503119-32e5a13c-f70c-4102-8257-e97b553d7708.png">

What you should do is go here: https://console.cloud.google.com/apis/credentials, and create your own credentials 

<img width="543" alt="image" src="https://user-images.githubusercontent.com/7578559/156503061-2c9094f4-1108-4a48-aa4c-305309b95162.png">

**Comment by samintokyo** [comment]
Hi!  Thanks so much - i have already done this and added that to the env.js
file but it still doesnt work./

**Comment by samintokyo** [comment]
<img width="1195" alt="Screen Shot 2022-03-03 at 15 07 03" src="https://user-images.githubusercontent.com/99787262/156506490-b405c693-499f-4e41-a05c-d400fe45c5da.png">
i get a bad request error after it redirects

It could have something to do with the database?

**Comment by lakesare** [comment]
I see, I get the same error.
I should be able to dive into this a bit later, I'll be on the planes in
the next few days. Could you please message me with this issue in a few
days if you don't manage to figure this out on your own in the meantime.

**Comment by samintokyo** [comment]
Thank you so much for your help!

I had my server hosting company install and launch the site so i think it
might be an issue with the database?

There was nothing in the database so i physically uploaded the sql scripts
that were contained in the database folder.  Not sure if that has anything
to do with it.

Also the jwt token was changed to an arbitrary value that i am not sure if
its correct.

Sorry to bother you but this is a good learning experience for me as it is
an area where im unsure of how to do many things
